src/v/test_utils/http_imposter.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "seastarx.h"
+#include "test_utils/registered_urls.h"
+#include "vassert.h"
+
+#include <seastar/core/sstring.hh>
+#include <seastar/http/httpd.hh>
+
+namespace internal {
+
+// Helper to progress over a range and check if the current element is present
+// in it. If the element is found at a position, the range for future searches
+// is adjusted to that position. Used for checking if a set of elements is
+// present in the same order in another searched range.
+//
+// eg [A B C] are present in [1 A C B X C D A] in order
+template<typename Iterator>
+struct search_state {
+    search_state(Iterator begin, Iterator end)
+      : _begin{begin}
+      , _end{end} {}
+
+    template<typename Url>
+    bool operator()(Url url) {
+        auto found = std::find_if(
+          _begin, _end, [&url](const auto& u) { return u._url == url; });
+        if (found == _end) {
+            return false;
+        }
+        _begin = found;
+        return true;
+    }
+
+private:
+    Iterator _begin;
+    Iterator _end;
+};
+} // namespace internal
+
+class http_imposter_fixture {
+public:
+    static constexpr std::string_view httpd_host_name = "127.0.0.1";
+    static constexpr uint httpd_port_number = 4430;
+
+public:
+    http_imposter_fixture();
+    ~http_imposter_fixture();
```

Comment:
other fixtures intend to inherit from this, so i'd mark this as virtual

Reply:
fixed